lib/Parser/Parser.cpp

```diff
@@ -1122,6 +1131,74 @@ bool Parser::parseLine(std::string &ErrStr) {
                   }
                   if (!consumeToken(ErrStr))
                     return false;
+                } else if (CurTok.str() == "range") {
+                  // look for =
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::Eq) {
+                    ErrStr = makeErrStr(TP, "expected '=' for range as 'range='");
+                    return false;
+                  }
+
+                  // look for [ (inclusive paren)
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::OpenBracket) {
+                    ErrStr = makeErrStr(TP, "expected '[' to specify lower bound of range");
+                    return false;
+                  }
+
+                  // look for untypedconst
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::UntypedInt && CurTok.K == Token::Int) {
+                    ErrStr = makeErrStr(TP, "expected lower bound of range without width");
+                    return false;
+                  }
+                  if (CurTok.K != Token::UntypedInt) {
+                    ErrStr = makeErrStr(TP, "expected lower bound of range");
+                    return false;
+                  }
+                  Lower = CurTok.Val;
+
+                  if (Lower.getBitWidth() != InstWidth) {
+                    Lower = Lower.sextOrTrunc(InstWidth);
+                  }
+
+                  // look for comma
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::Comma) {
+                    ErrStr = makeErrStr(TP, "expected ',' after lower bound of range");
+                    return false;
+                  }
+
+                  // look for untypedconst
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::UntypedInt && CurTok.K == Token::Int) {
```

Comment:
this check also doesn't seem helpful, I think it can be removed

Reply:
If you think, we don't need this kind of information, I can take it off completely. Otherwise, I was just going to simplify this error check as I mentioned above. IMO, this seems useful. Because, while I was writing the tests manually, I accidentally specified bitwidth against all constants that I used for the lower/upper bound. And, having such error message was helpful to figure out what's wrong in this.

Reply:
I have simplified this error check, but I still have two different errors: one to make sure user does not specify width with lower/upper bound, and second to make sure they do specify an untyped bound.

Reply:
Please check once, so that we can resolve this comment.